chore: optimize the magic number to Option and extract common code

1996fanrui opened the pull request:
Changes:

- Refactor the type of `rolling.State#next_date` from usize to `Option<usize>`
  - `rolling.State#next_date` is 0 when rotation is Rotation::Never. 
  - Using Option compared to magic number can intuitively tell developer that `next_date` is not effective.
- Move `date.unix_timestamp() as usize` inside of `next_date`
  - All callers of `Rotation#next_date` convert the data to timestamp, we can convert it inside of `Rotation#next_date`
  - Also, `Rotation#next_date` needs to rename to `Rotation#next_date_timestamp`

1996fanrui reviewed (commented):
Hey @tisonkun , would you mind taking a look this minor refactor in your free time? thank you very much.

tisonkun commented on src/append/rolling_file/rolling.rs:
```diff
@@ -320,12 +317,9 @@ impl State {
-        } else {
-            date.unix_timestamp() as usize >= next_date
-        }
+        self.next_date_timestamp
```
IIRC `Option` has an `is_some_and`  method to simplify these lines.

tisonkun replied:
Handled in https://github.com/tisonkun/logforth/pull/33/commits/2894986e323d5a4f9036fc79723c916f07948b60.

tisonkun reviewed (approved):
LGTM. One minor comment inline.

tisonkun commented:
Since we touch a few times the rolling file implementation, I'll appreciate it if you can try to port some tests in https://github.com/tokio-rs/tracing/blob/master/tracing-appender/src/rolling.rs so that we can guarantee the rolling behavior kept during changes.

tisonkun commented:
Thanks for your contribution!

1996fanrui commented:
Thanks for the quick review and merge!

Ahhh, I addressed your comment and I'm trying to add some teets on my Mac. 

I will create a new issue to add test later due to it's already merged.

tisonkun commented:
@1996fanrui OK. I may be too hurry to merge a PR :D
